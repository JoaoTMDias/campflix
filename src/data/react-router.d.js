import { array, arrayOf, bool, func, number, object, oneOf, oneOfType, shape, string } from "prop-types";

export const location = shape({
	hash: string.isRequired,
	key: string, // only in createBrowserHistory and createMemoryHistory
	pathname: string.isRequired,
	search: string.isRequired,
	state: oneOfType([array, bool, number, object, string]), // only in createBrowserHistory and createMemoryHistory
});

export const history = shape({
	action: oneOf(["PUSH", "REPLACE", "POP"]).isRequired,
	block: func.isRequired,
	canGo: func, // only in createMemoryHistory
	createHref: func.isRequired,
	entries: arrayOf(location), // only in createMemoryHistory
	go: func.isRequired,
	goBack: func.isRequired,
	goForward: func.isRequired,
	index: number, // only in createMemoryHistory
	length: number,
	listen: func.isRequired,
	location: location.isRequired,
	push: func.isRequired,
	replace: func.isRequired,
});

export const match = shape({
	isExact: bool,
	params: object.isRequired,
	path: string.isRequired,
	url: string.isRequired,
});

const routeShape = {
	path: string,
	exact: bool,
	strict: bool,
	sensitive: bool,
	component: func,
};
routeShape.routes = arrayOf(shape(routeShape));

export const route = shape(routeShape);

export default {
	location,
	history,
	match,
	route,
};
