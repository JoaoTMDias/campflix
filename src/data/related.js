import { MOVIE_DETAILS } from "../constants/index";

export const RELATED = {
  id: "803a5589-d36a-4e94-a821-1eb32539c896",
  title: "Top 10 Movies",
  data: [
    {
      id: "c19405b7-8fa8-47d7-a4af-63a5498100f7",
      name: "related-movies",
      to: `${MOVIE_DETAILS}/c19405b7-8fa8-47d7-a4af-63a5498100f7`,
      title: "Unbroken",
      background: {
        src:
          "https://occ-0-1289-358.1.nflxso.net/dnm/api/v6/0DW6CdE4gYtYx8iy3aj8gs9WtXE/AAAABbHsyrc2aTD1fcpczqYlsPeFuHNZUxvXiTSFN57MwlK1HLqSRl6XrLbciOMGQ7nBHSMvqVGM_1QhgJHpmCF-HWTolr4.webp?r=02b",
        alt: "Unbroken cover",
      },
      age: "16+",
      duration: "2h17m",
      tags: ["Violent", "Gritty", "Inspiring"],
    },
    {
      id: "9ef63bf8-86a1-423f-9c97-97cfc997e6a0",
      name: "related-movies",
      to: `${MOVIE_DETAILS}/9ef63bf8-86a1-423f-9c97-97cfc997e6a0`,
      title: "American Sniper",
      background: {
        src:
          "https://occ-0-3639-358.1.nflxso.net/dnm/api/v6/0DW6CdE4gYtYx8iy3aj8gs9WtXE/AAAABW1rFTbflGPzSRLVKVmHh0YY0LjaZx6kedCe_E_7GFs33LyMi88mLZNEmWwqsWfquR8lDPV9hAtopMnz8ET_ug7IIL0.webp?r=356",
        alt: "American Sniper cover",
      },
      age: "16+",
      duration: "1h30m",
      tags: ["Gritty", "Action & Adventure", "Drama"],
    },
    {
      id: "208bf44e-c12f-4d9c-895b-1c83b91ac274",
      name: "related-movies",
      to: `${MOVIE_DETAILS}/208bf44e-c12f-4d9c-895b-1c83b91ac274`,
      title: "Saving Private Ryan",
      background: {
        src:
          "https://occ-0-1289-358.1.nflxso.net/dnm/api/v6/0DW6CdE4gYtYx8iy3aj8gs9WtXE/AAAABR-HeH_i642HmC1UCjr-lwmiXBBUeHQ0wIC5XMW7oaVPiySg8ZJKXdLRZZMphlTiKVReTWIwULDd9rAfn_c2VwPgOV0.webp?r=7dc",
        alt: "Saving Private Ryan cover",
      },
      age: "16+",
      duration: "1h30m",
      tags: ["Rousing", "Violent", "Gritty"],
    },
    {
      id: "3ef96398-8109-4903-832b-99ffdfc221a4",
      name: "related-movies",
      to: `${MOVIE_DETAILS}/3ef96398-8109-4903-832b-99ffdfc221a4`,
      title: "The Theory of Everything",
      background: {
        src:
          "https://occ-0-1289-358.1.nflxso.net/dnm/api/v6/0DW6CdE4gYtYx8iy3aj8gs9WtXE/AAAABeXT9WO5RHVjCFTDfqJB81AqNmo4k3qclnna-LqZxhWSG15dMRKQfdY37uVFebKswDCKuEKDW0PmLpP0H1RcqM3pMrM.webp?r=0ef",
        alt: "The Theory of Everything cover",
      },
      age: "13+",
      duration: "2h03m",
      tags: ["Inspiring", "Emotional", "Romantic"],
    },
    {
      id: "06e9459d-0f86-4efa-8bd3-7cf8bc3fe279",
      name: "related-movies",
      to: `${MOVIE_DETAILS}/06e9459d-0f86-4efa-8bd3-7cf8bc3fe279`,
      title: "The Spy Episode 4",
      background: {
        src:
          "https://occ-0-1289-358.1.nflxso.net/dnm/api/v6/0DW6CdE4gYtYx8iy3aj8gs9WtXE/AAAABadOu753W5axKpsw-LChNmg75SAcEIJk1gPahkgZM6xYQKyfoIp4L8xvXhFaOIKCTOBv5BSw_mWTEuchOhiMrFBTjKPdoOtUDZI7DYAj8_Y96C0_4bsuuJPX9T9B.jpg?r=f69",
        alt: "The Spy Episode 4 cover",
      },
      age: "16+",
      duration: "56m",
      tags: ["Inspiring", "Emotional", "Romantic"],
    },
    {
      id: "ef4c5a34-5930-4c85-b3d0-3992b38dffd8",
      name: "related-movies",
      to: `${MOVIE_DETAILS}/ef4c5a34-5930-4c85-b3d0-3992b38dffd8`,
      title: "The Departed",
      background: {
        src:
          "https://occ-0-1289-358.1.nflxso.net/dnm/api/v6/0DW6CdE4gYtYx8iy3aj8gs9WtXE/AAAABeSFyzAAKicKhzDKqfFKSVMMJnPM7VwBEsb3efKXFf6GrROcxVv_G4BouJfzi9lFByjAyJ4utHBWXjIKAUdQYyRNxCk.webp?r=679",
        alt: "The Departed cover",
      },
      age: "16+",
      duration: "2h31m",
      tags: ["Violent", "Dark", "Film Noir"],
    },
    {
      id: "7fa5924a-3107-480e-8073-b8c6086f8ff0",
      name: "related-movies",
      to: `${MOVIE_DETAILS}/7fa5924a-3107-480e-8073-b8c6086f8ff0`,
      title: "Good Will Hunting",
      background: {
        src:
          "https://occ-0-1289-358.1.nflxso.net/dnm/api/v6/0DW6CdE4gYtYx8iy3aj8gs9WtXE/AAAABdw6l_cIxomVRSPU5rz067LEKjPiHagFXmtxT1NuRYKpVIIfgFPUuhsFLWZB0MHAOrKwYOunnWZMcam5kup8xuJ5fbA.webp?r=b13",
        alt: "Good Will Hunting cover",
      },
      age: "13+",
      duration: "2h06m",
      tags: ["Sentimental", "Heartfelt", "Drama"],
    },
    {
      id: "927568e9-1418-42e9-95e7-f233672d75fa",
      name: "related-movies",
      to: `${MOVIE_DETAILS}/927568e9-1418-42e9-95e7-f233672d75fa`,
      title: "The Departed",
      background: {
        src:
          "https://occ-0-1289-358.1.nflxso.net/dnm/api/v6/0DW6CdE4gYtYx8iy3aj8gs9WtXE/AAAABeSFyzAAKicKhzDKqfFKSVMMJnPM7VwBEsb3efKXFf6GrROcxVv_G4BouJfzi9lFByjAyJ4utHBWXjIKAUdQYyRNxCk.webp?r=679",
        alt: "The Departed cover",
      },
      age: "16+",
      duration: "2h31m",
      tags: ["Violent", "Dark", "Film Noir"],
    },
    {
      id: "4573d732-087f-4e15-8a97-10e5f4a95b11",
      name: "related-movies",
      to: `${MOVIE_DETAILS}/4573d732-087f-4e15-8a97-10e5f4a95b11`,
      title: "The Revenant",
      background: {
        src:
          "https://occ-0-1289-358.1.nflxso.net/dnm/api/v6/0DW6CdE4gYtYx8iy3aj8gs9WtXE/AAAABfLOEoF8rpya60LlF0csnEA4FopQSG7WchU6sFhsv_Cs3jAGXz-Vdi_zA0EzNApMWLccjpSWe0MYHsysAl3w2QMr__U.webp?r=394",
        alt: "The Revenant cover",
      },
      age: "16+",
      duration: "2h36m",
      tags: ["Violent", "Gritty", "Period Piece"],
    },
    {
      id: "24697ca6-2f61-4fef-ba4a-c03df768290a",
      name: "related-movies",
      to: `${MOVIE_DETAILS}/24697ca6-2f61-4fef-ba4a-c03df768290a`,
      title: "The Untouchables",
      background: {
        src:
          "https://occ-0-1289-358.1.nflxso.net/dnm/api/v6/0DW6CdE4gYtYx8iy3aj8gs9WtXE/AAAABdSLpU7vtWb9Uzu_HCJjtx1ATNHTFTzgzMp29mOcuJ-Od9ahBwB14yNpwmLhOMo4TbD-1TBoABqcIla5P8kDwwK7AVA.webp?r=c01",
        alt: "The Untouchables cover",
      },
      age: "16+",
      duration: "1h59m",
      tags: ["Forceful", "True Crime", "Action Thriller"],
    },
    {
      id: "7c0c43ce-a651-420c-a07a-73e8abc67cc8",
      name: "related-movies",
      to: `${MOVIE_DETAILS}/7c0c43ce-a651-420c-a07a-73e8abc67cc8`,
      title: "Black Hawk Down",
      background: {
        src:
          "https://occ-0-1289-358.1.nflxso.net/dnm/api/v6/0DW6CdE4gYtYx8iy3aj8gs9WtXE/AAAABe9S04IyXEDSyH8MwaX4IeM5ZmsyXMasO-Uc_lHcqB6W7_la_U0XGicau7_jDMmTctFMjlovhaWTd5cM7uYQ3Ve5oPE.webp?r=d71",
        alt: "Black Hawk Down cover",
      },
      age: "16+",
      duration: "2h24m",
      tags: ["Gory", "Action & Adventure", "Drama"],
    },
  ],
};
