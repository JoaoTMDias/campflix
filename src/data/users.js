import { HOMEPAGE_URL } from "../constants/index";
import ProfilePicJoao from "../assets/images/profile-1-joao.png";
import ProfilePicMickael from "../assets/images/profile-2-joao.png";
import ProfilePicLuis from "../assets/images/profile-3-joao.png";
import ProfilePicCelso from "../assets/images/profile-4-joao.png";

export const USERS = [
	{
		id: "0f04ab1c-1f33-4896-9a82-f08a206f7fbe",
		to: HOMEPAGE_URL,
		name: "João",
		image: {
			src: ProfilePicJoao,
			alt: "Profile picture",
		},
		label: "Choose João's profile",
	},
	{
		id: "31f7e325-801d-4305-bd5b-d2d29c8f2ef4",
		to: HOMEPAGE_URL,
		name: "Mickael",
		image: {
			src: ProfilePicMickael,
			alt: "Profile picture",
		},
		label: "Choose Mickael's profile",
	},
	{
		id: "f32dd5d6-6ad5-4469-b9cf-de6b6517d1f9",
		to: HOMEPAGE_URL,
		name: "Luís",
		image: {
			src: ProfilePicLuis,
			alt: "Profile picture",
		},
		label: "Choose Luis's profile",
	},
	{
		id: "462c053e-ec01-44b6-8fee-020dad249e2f",
		to: HOMEPAGE_URL,
		name: "Celso",
		image: {
			src: ProfilePicCelso,
			alt: "Profile picture",
		},
		label: "Choose Celso's profile",
	},
];
