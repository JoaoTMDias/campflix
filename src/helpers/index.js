export * from "./history";
export * from "./keycodes";
export * from "./get-genre-name";
