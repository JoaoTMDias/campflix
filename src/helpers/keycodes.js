/**
 * The copyright of this file belongs to Feedzai. The file cannot be
 * reproduced in whole or in part, stored in a retrieval system,
 * transmitted in any form, or by any means electronic, mechanical,
 * photocopying, or otherwise, without the prior permission of the owner.
 *
 * © 2019 Feedzai, Strictly Confidential
 */

/**
 * keyCodes
 *
 * Contains the values for the keyCode property in keyboard events
 *
 * @author João Dias (joao.dias@feedzai.com)
 */
export const KEYCODES = {
  BACKSPACE_DELETE: 8,
  TAB: 9,
  CLEAR: 12,
  ENTER: 13,
  OK: 13,
  SHIFT: 16,
  CONTROL: 17,
  ALT: 18,
  PAUSE_BREAK: 19,
  CAPS_LOCK: 20,
  ESC: 27,
  SPACE: 32,
  PAGE_UP: 33,
  PAGE_DOWN: 34,
  END: 35,
  HOME: 36,
  ARROW_LEFT: 37,
  ARROW_UP: 38,
  ARROW_RIGHT: 39,
  ARROW_DOWN: 40,
  ZERO: 0,
  ONE: 49,
  TWO: 50,
  THREE: 51,
  FOUR: 52,
  FIVE: 53,
  SIX: 54,
  SEVEN: 55,
  EIGHT: 56,
  NINE: 57,
};

export default KEYCODES;
