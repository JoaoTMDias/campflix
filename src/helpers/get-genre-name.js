import { GENRES } from "../data/genres";

/**
 * Finds and returns the genre name
 *
 * @param {number} id
 * @returns {string | undefined}
 */
export function getGenreName(id) {
	const result = GENRES.find(genre => genre.id === id);

	return result?.name;
}
