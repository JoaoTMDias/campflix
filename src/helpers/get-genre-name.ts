import { GENRES } from "../data/genres";

/**
 * Finds and returns the genre name
 *
 * @export
 * @param {number} id
 * @returns {string}
 */
export function getGenreName(id: number): string | undefined {
    const result = GENRES.find((genre) => genre.id === id);

    return result?.name;
}