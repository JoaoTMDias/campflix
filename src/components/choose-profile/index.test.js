import React from "react";
import { Router } from "react-router";
import { history } from "../../helpers";
import { render, cleanup } from "@testing-library/react";
import { USERS } from "../../data/users";
import ChooseProfile from "./index";

afterEach(cleanup);

const ChooseProfileWithRouter = () => {
  return (
    <Router history={history}>
      <ChooseProfile />
    </Router>
  );
};

describe("<ChooseProfile />", () => {
  it("renders without error", () => {
    const component = render(<ChooseProfileWithRouter />);

    expect(component).toMatchSnapshot();
  });

  it("renders four profiles", async () => {
    const { getAllByTestId } = render(<ChooseProfileWithRouter />);

    const profiles = await getAllByTestId("profile-link");

    expect(profiles.length).toBe(USERS.length);
  });
});
