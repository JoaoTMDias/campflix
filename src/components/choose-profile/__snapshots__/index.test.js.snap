// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChooseProfile /> renders without error 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <article
        class="choose-profile"
        id="choose-profile"
      >
        <h2>
          Who's watching?
        </h2>
        <nav
          class="choose-profile__content"
        >
          <ul
            class="choose-profile__buttons"
          >
            <a
              aria-label="Choose João's profile"
              class="profile-link"
              data-testid="profile-link"
              href="/home"
              id="0f04ab1c-1f33-4896-9a82-f08a206f7fbe"
            >
              <figure
                class="profile-link__figure"
              >
                <img
                  alt="Profile picture"
                  class="profile-link__image"
                  height="200"
                  src="profile-1-joao.png"
                  width="200"
                />
              </figure>
              <h3
                class="profile-link__name"
              >
                João
              </h3>
            </a>
            <a
              aria-label="Choose Mickael's profile"
              class="profile-link"
              data-testid="profile-link"
              href="/home"
              id="31f7e325-801d-4305-bd5b-d2d29c8f2ef4"
            >
              <figure
                class="profile-link__figure"
              >
                <img
                  alt="Profile picture"
                  class="profile-link__image"
                  height="200"
                  src="profile-2-joao.png"
                  width="200"
                />
              </figure>
              <h3
                class="profile-link__name"
              >
                Mickael
              </h3>
            </a>
            <a
              aria-label="Choose Luis's profile"
              class="profile-link"
              data-testid="profile-link"
              href="/home"
              id="f32dd5d6-6ad5-4469-b9cf-de6b6517d1f9"
            >
              <figure
                class="profile-link__figure"
              >
                <img
                  alt="Profile picture"
                  class="profile-link__image"
                  height="200"
                  src="profile-3-joao.png"
                  width="200"
                />
              </figure>
              <h3
                class="profile-link__name"
              >
                Luís
              </h3>
            </a>
            <a
              aria-label="Choose Celso's profile"
              class="profile-link"
              data-testid="profile-link"
              href="/home"
              id="462c053e-ec01-44b6-8fee-020dad249e2f"
            >
              <figure
                class="profile-link__figure"
              >
                <img
                  alt="Profile picture"
                  class="profile-link__image"
                  height="200"
                  src="profile-4-joao.png"
                  width="200"
                />
              </figure>
              <h3
                class="profile-link__name"
              >
                Celso
              </h3>
            </a>
          </ul>
        </nav>
      </article>
    </div>
  </body>,
  "container": <div>
    <article
      class="choose-profile"
      id="choose-profile"
    >
      <h2>
        Who's watching?
      </h2>
      <nav
        class="choose-profile__content"
      >
        <ul
          class="choose-profile__buttons"
        >
          <a
            aria-label="Choose João's profile"
            class="profile-link"
            data-testid="profile-link"
            href="/home"
            id="0f04ab1c-1f33-4896-9a82-f08a206f7fbe"
          >
            <figure
              class="profile-link__figure"
            >
              <img
                alt="Profile picture"
                class="profile-link__image"
                height="200"
                src="profile-1-joao.png"
                width="200"
              />
            </figure>
            <h3
              class="profile-link__name"
            >
              João
            </h3>
          </a>
          <a
            aria-label="Choose Mickael's profile"
            class="profile-link"
            data-testid="profile-link"
            href="/home"
            id="31f7e325-801d-4305-bd5b-d2d29c8f2ef4"
          >
            <figure
              class="profile-link__figure"
            >
              <img
                alt="Profile picture"
                class="profile-link__image"
                height="200"
                src="profile-2-joao.png"
                width="200"
              />
            </figure>
            <h3
              class="profile-link__name"
            >
              Mickael
            </h3>
          </a>
          <a
            aria-label="Choose Luis's profile"
            class="profile-link"
            data-testid="profile-link"
            href="/home"
            id="f32dd5d6-6ad5-4469-b9cf-de6b6517d1f9"
          >
            <figure
              class="profile-link__figure"
            >
              <img
                alt="Profile picture"
                class="profile-link__image"
                height="200"
                src="profile-3-joao.png"
                width="200"
              />
            </figure>
            <h3
              class="profile-link__name"
            >
              Luís
            </h3>
          </a>
          <a
            aria-label="Choose Celso's profile"
            class="profile-link"
            data-testid="profile-link"
            href="/home"
            id="462c053e-ec01-44b6-8fee-020dad249e2f"
          >
            <figure
              class="profile-link__figure"
            >
              <img
                alt="Profile picture"
                class="profile-link__image"
                height="200"
                src="profile-4-joao.png"
                width="200"
              />
            </figure>
            <h3
              class="profile-link__name"
            >
              Celso
            </h3>
          </a>
        </ul>
      </nav>
    </article>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
